src/jit/importer.cpp

```diff
@@ -11226,6 +11226,20 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 #endif // _TARGET_64BIT_
                 }
 
+#ifdef FEATURE_CORECLR
+                if (IsOperArithmetic(oper))
+                {
+                    const typeInfo& tiOp1 = impStackTop(1).seTypeInfo;
+                    const typeInfo& tiOp2 = impStackTop().seTypeInfo;
+
+                    // check for structure too because typeof(ldloca struct) is struct
```

Comment:
Actually DevDiv_278523 test case tests exactly you are asking, (ldloca(struct) + ldc.i8)